File: java/Day_2/src/Day_2.java
import java.util.Scanner;

/**
 * Problème
 * --------
 * Concevoir un programme et un ordinogramme qui calcule la somme de deux nombres.
 * Le programme est protégé par un mot de passe 'constantine'
 *
 * Catégorie
 * ---------
 * Numbers
 *
 * Niveau
 * ------
 * Débutant
 *
 * @author 10Glo (https://github.com/10Glo)
 * @version 1.0
 * @since 16/09/2025
 */

public class Day_2 {

    public static void main(String[] args) {

        // --- Déclaration des Variables ---
        Scanner scanner = new Scanner(System.in);
        int firstNumber = 0;
        int secondNumber = 0;
        String password = "";

        // --- Titre du projet ---
        System.out.println("==============================================");
        System.out.println(":::::::::      :::   :::   :::      ::::::::  \n" +
                ":+:    :+:   :+: :+: :+:   :+:     :+:    :+: \n" +
                "+:+    +:+  +:+   +:+ +:+ +:+            +:+  \n" +
                "+#+    +:+ +#++:++#++: +#++:           +#+    \n" +
                "+#+    +#+ +#+     +#+  +#+          +#+      \n" +
                "#+#    #+# #+#     #+#  #+#         #+#       \n" +
                "#########  ###     ###  ###        ########## ");
        System.out.println("==============================================\n");

        // --- Ordinogramme ---
        System.out.println("\uD83D\uDD37    Organigramme de Programmation \n\n" +
                "                           [Début]\n" +
                "                              ↓\n" +
                "                       [Lire password]\n" +
                "                              ↓\n" +
                "           [password isEqual() 'constantine' ?]◄──\n" +
                "                    / Oui             Non \\\n" +
                "                   ↓                       ↓\n" +
                "   [Lire firstNumber & secondNumber]     [Fin]\n" +
                "               ↓\n" +
                "[firstNumber & secondNumber isNumber() ?]◄──\n" +
                "        / Oui             Non \\\n" +
                "       ↓\n" +
                "[Afficher Somme]"+
                "       ↓\n" +
                "     [Fin]\n");

        // --- Saisie du Mot de Passe ---
        System.out.println("\uD83D\uDD11️  Veuillez entrer le mot de passe :  ");
        password = scanner.nextLine();
        if (password.equals("constantine"))
            System.out.println("Le mot de passe est correct ✅\n");
        else {
            System.out.println("Le mot de passe n'est pas correct ❌");
            System.out.println("\nVeuillez Recommencer !!!");

            scanner.close();
            System.out.println("\nFin du programme ✅");
            System.exit(0);
        }


        // --- Saisie des informations ---
        try {
            System.out.print("➡️  Entrez le premier nombre : ");
            firstNumber = scanner.nextInt();

            System.out.print("➡️  Entrez le deuxième nombre : ");
            secondNumber = scanner.nextInt();
        } catch (Exception exception) {
            System.out.println("\n⚠️ Erreur : Vous devez entrer un nombre valide !\n");

            scanner.close();
            System.out.println("\nFin du programme ✅");
            System.exit(0);
        }

        // --- Résumé des informations ---
        System.out.println("\n-------------------------------------");
        System.out.println(" 📋 Résumé des informations saisies");
        System.out.println("-------------------------------------");
        System.out.println(" \uD83D\uDD22 Le premier nombre    : " + firstNumber);
        System.out.println(" \uD83D\uDD22 Le deuxième nombre    : " + secondNumber + "\n");

        // --- Résutats ---
        System.out.println("--------------");
        System.out.println(" 📋 Résutats");
        System.out.println("--------------\n");
        System.out.println("➡️  La somme de " + firstNumber + " et " + secondNumber + " est --> " + (firstNumber + secondNumber));

        // --- Message de fin ---
        scanner.close();
        System.out.println("Fin du programme ✅");
        System.out.println("-------------------------------------------");

    }

}

File: javascript/Day_2/Day_2.js
/**
 * Problème
 * --------
 * Concevoir un programme qui demande l'âge d'une personne.
 * Ensuite, l'informe de sa catégorie :
 * "Enfance : de 0 à 12 ans",
 * "Adolescence : de 12 à 18 ans",
 * "Âge Adulte : de 18 à 65 ans",
 * "Séniors ou Personnes âgées : À partir de 65 ans"
 *
 * Catégorie
 * ---------
 * Numbers
 *
 * Niveau
 * ------
 * Débutant
 *
 * @author 10Glo
 * @version 1.0
 * @since 16/09/2025
 */

const readline = require("readline");

// Création d'une interface readline
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
});

// --- Titre du projet ---
console.log("==============================================");
console.log(":::::::::      :::   :::   :::         :::     ");
console.log(":+:    :+:   :+: :+: :+:   :+:        :+:      ");
console.log("+:+    +:+  +:+   +:+ +:+ +:+        +:+ +:+   ");
console.log("+#+    +:+ +#++:++#++: +#++:        +#+  +:+   ");
console.log("+#+    +#+ +#+     +#+  +#+        +#+#+#+#+#+ ");
console.log("#+#    #+# #+#     #+#  #+#              #+#   ");
console.log("#########  ###     ###  ###              ###   ");
console.log("==============================================\n");

// --- Saisie de l'âge ---
rl.question("➡️  Entrez votre âge (Entrez juste le nombre) : ", (input) => {
    let age = parseInt(input);

    if (isNaN(age)) {
        console.log("\n⚠️ Erreur : Vous devez entrer un nombre valide !");
        console.log("\nFin du programme ✅");
        rl.close();
        process.exit(0);
    }

    // --- Résumé des informations ---
    console.log("\n-------------------------------------");
    console.log(" 📋 Résumé des informations saisies");
    console.log("-------------------------------------");
    console.log(" 🔢 L'âge que vous avez entré : " + age + " ans\n");

    // --- Résultats ---
    console.log("--------------");
    console.log(" 📋 Résultats");
    console.log("--------------");

    if (age < 0) {
        console.log("Votre âge ne peut pas être négatif.\n");
    } else if (age <= 12) {
        console.log("➡️  Votre catégorie est 'ENFANT'\n");
    } else if (age <= 18) {
        console.log("➡️  Votre catégorie est 'ADOLESCENCE'\n");
    } else if (age <= 65) {
        console.log("➡️  Votre catégorie est 'ADULTE'\n");
    } else {
        console.log("➡️  Votre catégorie est 'SÉNIORS'\n");
    }

    // --- Message de fin ---
    console.log("Fin du programme ✅");
    console.log("-------------------------------------------");

    rl.close();
});
